core/src/main/scala/filodb.core/memstore/OnDemandPagingShard.scala

```diff
@@ -104,7 +104,9 @@ TimeSeriesShard(dataset, storeConfig, shardNum, bufferMemoryManager, rawStore, m
             rawStore.readRawPartitions(dataset, allDataCols, multiPart, computeBoundingMethod(pagingMethods))
               // NOTE: this executes the partMaker single threaded.  Needed for now due to concurrency constraints.
               // In the future optimize this if needed.
-              .mapAsync { rawPart => partitionMaker.populateRawChunks(rawPart).executeOn(singleThreadPool) }
+              .mapAsync { rawPart =>
+                partitionMaker.populateRawChunks(rawPart).executeOn(singleThreadPool).asyncBoundary
```

Comment:
Can we add a comment explaining why asyncBoundary is needed?  Thx

Reply:
I already added in one place. I can add in all places.